src/player.cpp

```diff
@@ -12649,6 +12830,46 @@ int player::get_armor_cut_base(body_part bp) const
     return ret;
 }
 
+int player::get_armor_bash_bonus( body_part bp ) const
```

Comment:
The entire reason `bash_armor_bonus` and the like exist is so that you DON'T need to recalculate them every time you need them, only once per turn.
The effects that affect armor should be applied where other effects are - in `player::reset_stats()`.